预测Facebook签到位置

    1、项目描述：
        （1）目的：预测一个人将要签到的地方
        （2）虚拟环境：10公里*10公里，共100平方公里的约10万个地方
        （3）任务：根据用户的位置，准确性和时间戳等预测用户下一次签到的位置

    2、数据集介绍
        row_id:样本序号
        x:x轴坐标
        y:y轴坐标
        accuracy:准确性，定位精度
        time:时间戳
        place_id:签到位置，这也是需要预测的内容

    3、数据位置：
    Data/facebook-vpredicting-check-ins
    训练数据：train.csv
    测试数据：test.csv
    案例数据：sample_submission.csv

In [1]:
# 查看数据
import pandas as pd

train_data = pd.read_csv("../Data/facebook-vpredicting-check-ins/train.csv")
test_data = pd.read_csv("../Data/facebook-vpredicting-check-ins/test.csv")

In [2]:
train_data.head()

row_id       x       y  accuracy    time    place_id
0       0  0.7941  9.0809        54  470702  8523065625
1       1  5.9567  4.7968        13  186555  1757726713
2       2  8.3078  7.0407        74  322648  1137537235
3       3  7.3665  2.5165        65  704587  6567393236
4       4  4.0961  1.1307        31  472130  7440663949

In [3]:
test_data.head()

row_id       x       y  accuracy    time
0       0  0.1675  1.3608       107  930883
1       1  7.3909  2.5301        35  893017
2       2  8.0978  2.3473        62  976933
3       3  0.9990  1.0591        62  907285
4       4  0.6670  9.7254        40  914399

In [4]:
train_data.describe()

row_id             x             y      accuracy          time  \
count  2.911802e+07  2.911802e+07  2.911802e+07  2.911802e+07  2.911802e+07   
mean   1.455901e+07  4.999770e+00  5.001814e+00  8.284912e+01  4.170104e+05   
std    8.405649e+06  2.857601e+00  2.887505e+00  1.147518e+02  2.311761e+05   
min    0.000000e+00  0.000000e+00  0.000000e+00  1.000000e+00  1.000000e+00   
25%    7.279505e+06  2.534700e+00  2.496700e+00  2.700000e+01  2.030570e+05   
50%    1.455901e+07  5.009100e+00  4.988300e+00  6.200000e+01  4.339220e+05   
75%    2.183852e+07  7.461400e+00  7.510300e+00  7.500000e+01  6.204910e+05   
max    2.911802e+07  1.000000e+01  1.000000e+01  1.033000e+03  7.862390e+05   

           place_id  
count  2.911802e+07  
mean   5.493787e+09  
std    2.611088e+09  
min    1.000016e+09  
25%    3.222911e+09  
50%    5.518573e+09  
75%    7.764307e+09  
max    9.999932e+09

    3.步骤分析
        （1）对于数据做一些基本处理，这里所做的一些处理不一定达到很好的效果，我们只是简
            单测试，有些特征我们可以根据一些特征选择的方法进行处理.
                a、最小化数据集DataFrame.query()
                b、选取有用的时间特征
                c、将签到位置少于n个用户的删除
        （2）分割数据
        （3）标准化处理
        （4）k-近邻预测

    具体步骤：
        1.获取数据集 pd.read_csv()
        2.基本数据处理
            2.1 最小化数据集 DataFrame..query()
            2.2 选择时间特征
            2.3 去掉签到少于n个用户的地方
            2.4 确定特征值和目标值
            2.5 分割数据
        3.特征工程 -- 特征预处理（标准化）
        4.机器学习 -- knn + cv
        5.模型评估